Source/platform/locale.cpp

```diff
@@ -176,12 +176,12 @@ std::vector<std::string> GetLocales()
 
 	CFRelease(languages);
 #else
-	try {
-		std::string locale = std::locale("").name();
-		// strip off any encoding specifier, devX uses UTF8.
-		locales.emplace_back(locale.substr(0, locale.find('.')));
-	} catch (std::runtime_error &e) {
-		LogWarn("Locale detection failed: {}", e.what());
+	const string_view locale = setlocale(0, nullptr);
+	const string_view lang = locale.substr(0, locale.find('.'));

```

Comment:
This should probably just consider the messages (fallback to character type if not defined?) category instead of the whole locale. Guessing using LANGUAGE or LANG environment variables would be better here since locales are...almost useless from what I can see.
```suggestion
#ifdef LC_MESSAGES
	const string_view locale = setlocale(LC_MESSAGES, nullptr);
#else
	const string_view locale = setlocale(LC_CTYPE, nullptr);
#endif
	const string_view lang = locale.substr(0, locale.find('.'));
```
If a user has redefined part of their system locale `setlocale(0 /* LC_ALL*/, NULL)` gives a list of `key=value;key=value;...` pairs which will give a garbage first preference.

Having said that, don't bother changing it for this PR. I think it'd be better to rewrite this section to use `getenv(LANGUAGE) > getenv(LANG) > setlocal(LC_MESSAGES) > setlocale(LC_CTYPE)` as a preference order.

Reply:
Done